go/pkg/repository/s3.go

```diff
@@ -256,6 +256,28 @@ func (s *S3Repository) GetPathTar(tarPath, localPath string) error {
 	return extractTar(tmptarball, localPath)
 }
 
+func (s *S3Repository) GetPathItemTar(tarPath, itemPath, localPath string) error {
+	// archiver doesn't let us use readers, so download to temporary file
+	// TODO: make a better tar implementation
+	tmpdir, err := files.TempDir("tar")
+	if err != nil {
+		return err
+	}
+	defer os.RemoveAll(tmpdir)
+	tmptarball := filepath.Join(tmpdir, filepath.Base(tarPath))
+	if err := s.GetPath(tarPath, tmptarball); err != nil {
+		return err
+	}
+	exists, err := files.FileExists(tmptarball)
+	if err != nil {
+		return err
+	}
+	if !exists {
+		return &DoesNotExistError{msg: "GetPathTar: does not exist: " + tmptarball}
```

Comment:
let's not expose function name to end user.

Reply:
Done